libsolidity/ast/AST.cpp

```diff
@@ -213,11 +213,22 @@ vector<EventDefinition const*> const& ContractDefinition::interfaceEvents() cons
 					interfaceEvents.push_back(e);
 				}
 			}
-
 		return interfaceEvents;
 	});
 }
 
+vector<EventDefinition const*> const ContractDefinition::usedInterfaceEvents() const
+{
+	solAssert(annotation().creationCallGraph.set(), "");
+
+	std::vector<EventDefinition const*> events;
+
+	events += (*annotation().creationCallGraph)->emittedEvents;
+	events += (*annotation().deployedCallGraph)->emittedEvents;
+
+	return events;
```

Comment:
Why not
```
return
    ...->emittedEvents +
    ...->emittedEvents;
```
?

Reply:
emittedEvents is a `set`, so we can't just return the concatination